runelite-client/src/main/java/net/runelite/client/events/PartyJoinRequest.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, Jonathan Rousseau <https://github.com/JoRouss>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.events;
+
+import lombok.Value;
+
+@Value
+public class PartyJoinRequest
```

Comment:
What is the purpose of this? It seems like the party service could just use `DiscordJoinRequest` if it needed the join request info. I don't see any reason to try and decouple the party service from discord.

Reply:
I did it this way because they were already decoupled. This way, I can also keep track of which requests were sent through Discord and respond to PartyJoinReply accordingly. I know there can't be requests from any other plugins right now, but I wanted tokeep them decoupled so there can be in the future.

I don't mind coupling them if you want. I'll wait for your decision.

Reply:
This would be the result of coupling DiscordService to PartyPlugin
https://github.com/JoRouss/runelite/commit/585e280db53709659b1877c08e70f93fc39527fb